File: src/main/java/org/shark/example/service/quartz/DeleteJobService.java
package org.shark.example.service.quartz;

import lombok.RequiredArgsConstructor;
import org.quartz.JobKey;
import org.quartz.Scheduler;
import org.quartz.SchedulerException;
import org.quartz.TriggerKey;
import org.shark.example.service.base.pojo.ResponseDto;
import org.springframework.stereotype.Service;

@RequiredArgsConstructor
@Service
public class DeleteJobService {

    private final Scheduler scheduler;

    public ResponseDto<Void> start(String jobName, String jobGroupName) {
        try {
            TriggerKey key = TriggerKey.triggerKey(jobName, jobGroupName);
            scheduler.pauseTrigger(key);
            scheduler.unscheduleJob(key);
            scheduler.deleteJob(new JobKey(jobName, jobGroupName));
        } catch (SchedulerException e) {
            throw new RuntimeException(e);
        }

        return ResponseDto.<Void>builder()
                .status(true).build();
    }
}

File: src/main/kotlin/org/shark/example/service/quartz/DeleteJobService.kt
package org.shark.example.service.quartz

import org.quartz.JobKey
import org.quartz.Scheduler
import org.quartz.SchedulerException
import org.quartz.TriggerKey
import org.shark.example.service.base.pojo.ResponseDto
import org.springframework.stereotype.Service

@Service
class DeleteJobService(
    private val scheduler: Scheduler
) {
    fun start(jobName: String, jobGroupName: String): ResponseDto<Void> {
        try {
            val key = TriggerKey.triggerKey(jobName, jobGroupName)
            scheduler.pauseTrigger(key)
            scheduler.unscheduleJob(key)
            scheduler.deleteJob(JobKey(jobName, jobGroupName))
        } catch (e: SchedulerException) {
            throw RuntimeException(e)
        }

        return ResponseDto.builder<Void>()
            .status(true).build()
    }
}
